contracts/suilend/sources/obligation.move

```diff
@@ -260,6 +289,126 @@ module suilend::obligation {
+        let mut allowed_borrow_value_usd = decimal::from(0);
+        let mut unhealthy_borrow_value_usd = decimal::from(0);
+
+        while (i < vector::length(&obligation.deposits)) {
+            let deposit = vector::borrow_mut(&mut obligation.deposits, i);
+
+            let deposit_reserve = vector::borrow_mut(reserves, deposit.reserve_array_index);
+
+            reserve::compound_interest(deposit_reserve, clock);
+            reserve::assert_price_is_fresh(deposit_reserve, clock);
+
+            let market_value = reserve::ctoken_market_value(
+                deposit_reserve,
+                deposit.deposited_ctoken_amount
+            );
+            let market_value_lower_bound = reserve::ctoken_market_value_lower_bound(
+                deposit_reserve,
+                deposit.deposited_ctoken_amount
+            );
+
+            deposit.market_value = market_value;
+            deposited_value_usd = add(deposited_value_usd, market_value);
+
+            let (open_ltv, close_ltv) = get_ltvs(
+                obligation,
+                deposit_reserve,
+                is_emode,
+            );
+
+            allowed_borrow_value_usd = add(
+                allowed_borrow_value_usd,
+                mul(
+                    market_value_lower_bound,
+                    open_ltv,
+                ),
+            );
+
+            unhealthy_borrow_value_usd = add(
+                unhealthy_borrow_value_usd,
+                mul(
+                    market_value,
+                    close_ltv,
+                ),
+            );
+
+            i = i + 1;
+        };
+
+        obligation.deposited_value_usd = deposited_value_usd;
+        obligation.allowed_borrow_value_usd = allowed_borrow_value_usd;
+        obligation.unhealthy_borrow_value_usd = unhealthy_borrow_value_usd;
+
+        let mut i = 0;
+        let mut unweighted_borrowed_value_usd = decimal::from(0);
+        let mut weighted_borrowed_value_usd = decimal::from(0);
+        let mut weighted_borrowed_value_upper_bound_usd = decimal::from(0);
+        let mut borrowing_isolated_asset = false;
+
+        while (i < vector::length(&obligation.borrows)) {
+            let borrow = vector::borrow_mut(&mut obligation.borrows, i);
+
+            let borrow_reserve = vector::borrow_mut(reserves, borrow.reserve_array_index);
+            reserve::compound_interest(borrow_reserve, clock);
+            reserve::assert_price_is_fresh(borrow_reserve, clock);
+
+            compound_debt(borrow, borrow_reserve);
+
+            let market_value = reserve::market_value(borrow_reserve, borrow.borrowed_amount);
+            let market_value_upper_bound = reserve::market_value_upper_bound(
+                borrow_reserve, 
+                borrow.borrowed_amount
+            );
+
+            borrow.market_value = market_value;
+            unweighted_borrowed_value_usd = add(unweighted_borrowed_value_usd, market_value);
+
+            let borrow_weight = if (is_emode) {
```

Comment:
i think even if obligation is emode, if there's no corresponding emode entry, then the borrow weight has to be the one stored on the reserve (ie not 1). 

example: say user is depositing usdc and borrowing FUD. fud has a borrow weight of 2. the (usdc, fud) pair does not have a corresponding emode entry. then, if the user sets their obligation to emode, then the borrow weight being used is 1, which lets them borrow more fud than we intended.

Reply:
Good point. Refactored